crates/data-transformer/src/sierra_abi/complex_types.rs

```diff
@@ -0,0 +1,368 @@
+use super::data_representation::{
+    AllowedCalldataArgument, CalldataEnum, CalldataStruct, CalldataStructField, CalldataTuple,
+};
+use anyhow::{bail, ensure, Context, Result};
+use cairo_lang_parser::utils::SimpleParserDatabase;
+use cairo_lang_syntax::node::ast::{
+    Expr, ExprFunctionCall, ExprListParenthesized, ExprPath, ExprStructCtorCall,
+    OptionStructArgExpr, PathSegment, PathSegment::Simple, StructArg,
+};
+use cairo_lang_syntax::node::{Terminal, Token, TypedSyntaxNode};
+use itertools::Itertools;
+use starknet::core::types::contract::{AbiEntry, AbiEnum, AbiNamedMember, AbiStruct};
+use std::collections::HashSet;
+
+use super::parsing::{parse_argument_list, parse_expression};
+use super::{build_representation, SupportedCalldataKind};
+
+pub trait EnumOrStruct {
+    const VARIANT: &'static str;
+    const VARIANT_CAPITALIZED: &'static str;
+    fn name(&self) -> String;
+}
+
+impl EnumOrStruct for AbiStruct {
+    const VARIANT: &'static str = "struct";
+    const VARIANT_CAPITALIZED: &'static str = "Struct";
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+}
+
+impl EnumOrStruct for AbiEnum {
+    const VARIANT: &'static str = "enum";
+    const VARIANT_CAPITALIZED: &'static str = "Enum";
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+}
+
+fn validate_path_argument(
+    param_type: &str,
+    path_argument: &[String],
+    path_argument_joined: &String,
+) -> Result<()> {
+    if *path_argument.last().unwrap() != param_type.split("::").last().unwrap()
+        && path_argument_joined != param_type
```

Comment:
It shouldn't, because then it would not be possible to pass shorter paths  -e.g. `--calldata Enum::Variant` to function that expects `contract_name::Enum` argument